bugRzilla_review.Rmd

```diff
@@ -698,28 +700,99 @@ total_data <- total_data %>%
 ```
 
 ```{r total_data datatable}
-datatable(head(total_data, 13), options = list(scrollX = TRUE))
+# created bugs per year
+created_year_df <- data.frame(total_data$bug_id, 
+                              total_data$creation_year)
+created_year_df <- aggregate(total_data.bug_id~., 
+                             created_year_df, 
+                             function(x) length(unique(x)))
+colnames(created_year_df) <- c("creation_year", "Bug_count")
+
+# last modified bugs per year
+lastdiffed_year_df <- data.frame(total_data$bug_id, 
+                                 total_data$lastdiffed_year)
+lastdiffed_year_df <- aggregate(total_data.bug_id~., 
+                                lastdiffed_year_df, 
+                                function(x) length(unique(x)))
+colnames(lastdiffed_year_df) <- c("lastdiffed_year", "Bug_count")
+
+# created and last modified bugs per year in single dataframe
+lastdiffed_year_df <- pad(lastdiffed_year_df)
+cre_last_year <- data.frame(created_year_df, 
+                            lastdiffed_year_df)
+
+skim(cre_last_year)
+
+# created bugs per month
+created_month_df <- data.frame(total_data$bug_id, 
+                               total_data$creation_month)
+created_month_df <- aggregate(total_data.bug_id~., 
+                              created_month_df,
+                              function(x) length(unique(x)))
+colnames(created_month_df) <- c("creation_month", "Bug_count")
+
+# last modified bugs per month
+lastdiffed_month_df <- data.frame(total_data$bug_id, 
+                                  total_data$lastdiffed_month)
+lastdiffed_month_df <- aggregate(total_data.bug_id~., 
+                                 lastdiffed_month_df, 
+                                 function(x) length(unique(x)))
+colnames(lastdiffed_month_df) <- c("lastdiffed_month", "Bug_count")
+
+# created and last modified bugs per month in single dataframe
+lastdiffed_month_df <- pad(lastdiffed_month_df)
+cre_last_month <- data.frame(created_month_df, 
+                             lastdiffed_month_df)
+skim(cre_last_month)
+
+# last modified bugs per week
+created_week_df <- data.frame(total_data$bug_id, 
+                              total_data$creation_week)
+created_week_df <- aggregate(total_data.bug_id~., 
+                             created_week_df, 
+                             function(x) length(unique(x)))
+colnames(created_week_df) <- c("creation_week", "Bug_count")
+
+# last modified bugs per week
+lastdiffed_week_df <- data.frame(total_data$bug_id, 
+                                 total_data$lastdiffed_week)
+lastdiffed_week_df <- aggregate(total_data.bug_id~., 
+                                lastdiffed_week_df, 
+                                function(x) length(unique(x)))
+colnames(lastdiffed_week_df) <- c("lastdiffed_week", "Bug_count")
+
+# created and last modified bugs per week in single dataframe
+created_week_df <- pad(created_week_df)
+lastdiffed_week_df <- pad(lastdiffed_week_df)
```

Comment:
What does the pad function do? According to [the help page](https://search.r-project.org/CRAN/refmans/padr/html/pad.html) it fills the data with some values. However, is this really what we need here? It is normal to have bugs that are only written and not edited.

Reply:
The `pad()` function adds NA values where data is missing. The `pad()` function is needed as while merging the data frames the length of the data frames was not equal. I've tried the `merge()`, `cbind()` but the issue was not resolved. So to make them equal the pad function is used.

Reply:
How are the data.frames incompatible? If the data is missing how is it encoded on the data frame? Why didn't you mention this to me before?

Reply:
The **shape** of the data frames is incompatible as I'm merging two different columns in a single dataframe the length of the columns is not the same i.e. `created_month_df`, `lastdiffed_month_df` have `274` & `272` resepectively.